### PR TITLE
feat: add installer.sh w/ install/upgrade/uninstall options

### PR DESCRIPTION
**Pull request by Lanture1064**
`cd scripts`
`$ ./installer.sh [PARAM]`

--all = install all
--baas = install fabric-operator
--u4a = install cluster-component & u4a-component
--up-all = upgrade all
--up-baas = upgrade fabric-operator
--rm-baas = uninstall fabric-operator

### CODE REVIEW

**Comment by Lanture1064** [comment]
It's basically modified from e2e.sh, so I haven't integrate kind.sh yet

File: scripts/installer.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+kubeProxyNodeIP=$(kubectl get node ${kubeProxyNode} -owide | grep -v "NAME" | awk '{print $6}')
+
+echo "node info:"
+kubectl get node -o wide
+echo "ingressNodeIp is: ${ingressNodeIP}"
+echo "kubeProxyNodeIP is: ${kubeProxyNodeIP}"
+export ingressNodeIP=${ingressNodeIP}
+
+# step 3. repalce ingress node name
+cat ../u4a-component/charts/cluster-component/values.yaml | sed "s/<replaced-ingress-node-name>/${ingressNode}/g" \
```
Comment (0xff-dev) [inline scripts/installer.sh:96]: I think it should be changed to absolute path here.

```bash
ABSPATH=$(cd "$(dirname "$0")";pwd)
cat $ABSPATH/u4a-component/xxxx
```

**Comment by Abirdcfly** [comment]
This is a great usability change. And this is your first pull request in bestchains project. Thanks for your contribution! bestchains will be better because of you.

I guess it would be good to modify `e2e.sh` directly or delete `e2e.sh` so that we don't need to maintain two similar scripts at the same time.

and there is another point of concern here, fabric-operator's example-test  https://github.com/bestchains/fabric-operator/blob/b35f7a3cc27aabaa7472173f46afdf11400f6b29/config/samples/example-test.sh#L122-L137 also depends on these scripts. The modification here should not affect fabric-operator's example-test, or modify fabric-operator's example-test after the current pr merged.

File: scripts/installer.sh (hunk 1 above)
Comment (Lanture1064) [reply scripts/installer.sh:96]: The origin `e2e.sh` uses absolute path to co-operate with the `makefile` by `$ make e2e`.  
Maybe adding a similar `make installer [PARAM]` in the `makefile` will solve it ?

**Comment by Lanture1064** [comment]
#6 now merged into `e2e.sh` & using ABSPATH

**Review by bjwswang — CHANGES_REQUESTED** [review]
Pls place all changes in one commit.

**Review by Abirdcfly — CHANGES_REQUESTED** [review]
The GitHub Action should be modified to make the tests pass.

**Comment by Lanture1064** [comment]
Changes merged into one commit. Should every PR be created with one change commit like this way?

**Comment by bjwswang** [comment]
> Changes merged into one commit. Should every PR be created with one change commit like this way?

That depends on the purpose that you made this PR for... ^_^

**Review by bjwswang — CHANGES_REQUESTED** [review]
pls fix conlicts firstly

**Comment by bjwswang** [comment]
@Abirdcfly @Lanture1064 please work together to fix this conflicts

**Comment by Lanture1064** [comment]
It is now compatible with base branch & workflow check

**Pull request merged**